### ISSUE TITLE
about the data used for training

### ISSUE THREAD

**Issue by yinhongwei4079** [calico/basenji#95]
Hi @davek44 
 we have ATAC-seq data in different tissues and we want use the basenji to predict the activaty of snps. We wonder we put all the ATAC-seq in different tissues togother to the bsenji for trainning or just one tissue ?

**Comment by davek44** [comment]
Hi, I typically train multi-task. You can use basenji_data.py to create tfrecords for your data from bigwig files.

**Comment by yinhongwei4079** [comment]
@davek44 Thanks for your kindly reply! We multi epigenetic signals for one tissue like ATAC,CTCF,H3K27ac  and so on. can we put these epigenetic signals togother to the basenji for training and predicting?

**Comment by davek44** [comment]
The SNPs that overlap open chromatin regions are far more likely to be influential, so that’s a reasonable filter. However, sometimes variants outside of the peaks can create a new binding site, gaining function, and prove interesting, too. Proceeding either way seems fine to me.

**Comment by yinhongwei4079** [comment]
@davek44 all we want to know if we put all epigenetic data(ATAC_SEQ and four histone marker) to train the model by basenji ,is it reasonable to predict the effects of snps across the genome,special snps in noncoding regions?

**Comment by davek44** [comment]
Yes, it's reasonable

**Comment by yinhongwei4079** [comment]
i have a question about the file of heart_wigs.txt in the example of basenji_data.py.  we have seen sum_stat have two states: mean and sum. we didn't know which situation we give. we have two replicates of ATAC-seq in one tissue and the sum_stat is sum or mean, if we used for other chip-seq which item we can choose. we also didn't know the mean of clip , and why you give the value of 384, could you give us some guidline?

**Comment by yinhongwei4079** [comment]
@davek44 as you see we want to fullfil it in pigs but in pigs we didn't have files of GAPS_HUMAN=$$HG38/assembly/ucsc/hg38_gaps.bed
UMAP_HUMAN=$$HG38//mappability/umap_k24_t10_l32.bed
BLACK_HUMAN=$$HG38/blacklist/hg38.blacklist.rep.bed
could you gives us some guildilines for preparing the three files for pigs ?

**Comment by yinhongwei4079** [comment]
@davek44 we have put ATAC-seq and four histone markers to train and index file were 
index	identifier	file	clip	sum_stat	description
0	ATAC_Stomach_P348	/home/yinhongwei/pig_yin/epigenetic/ATAC_Stomach_P348.bw	384	mean	Stomach
1	ATAC_Stomach_P350	/home/yinhongwei/pig_yin/epigenetic/ATAC_Stomach_P350.bw	384	mean	Stomach
2	H3K27ac_Stomach_P348	/home/yinhongwei/pig_yin/epigenetic/H3K27ac_Stomach_P348.bw	384	mean	Stomach
3	H3K27ac_Stomach_P350	/home/yinhongwei/pig_yin/epigenetic/H3K27ac_Stomach_P350.bw	384	mean	Stomach
4	H3K27me3_Stomach_P348	/home/yinhongwei/pig_yin/epigenetic/H3K27me3_Stomach_P348.bw	384	mean	Stomach
5	H3K27me3_Stomach_P350	/home/yinhongwei/pig_yin/epigenetic/H3K27me3_Stomach_P350.bw	384	mean	Stomach
6	H3K4me1_Stomach_P348	/home/yinhongwei/pig_yin/epigenetic/H3K4me1_Stomach_P348.bw	384	mean	Stomach
7	H3K4me1_Stomach_P350	/home/yinhongwei/pig_yin/epigenetic/H3K4me1_Stomach_P350.bw	384	mean	Stomach
8	H3K4me3_Stomach_P348	/home/yinhongwei/pig_yin/epigenetic/H3K4me3_Stomach_P348.bw	384	mean	Stomach
9	H3K4me3_Stomach_P350	/home/yinhongwei/pig_yin/epigenetic/H3K4me3_Stomach_P350.bw	384	mean	Stomach
then we test the precision of model, but only three file(t0, t1, t2)were generated for AUROC. could you tell us why?

**Comment by yinhongwei4079** [comment]
@davek44 hi，we had trained our data for models and to predict the effects of snp with the script of basenji_sad.py. we have choosed 10million snps for predicting. but it was low as it is running for three days. could you give us some guidline to speed u the process??

**Comment by davek44** [comment]
To your first question, 'mean' and 'sum' refer to whether to take the mean or the sum of the nucleotide-annotated coverage in your BigWig files. If the values represent counts, I recommend 'sum'. If the values are more processed, e.g. log fold change, then I recommend 'mean'. 'clip' will simply clip the values to enforce a maximum. I use this to avoid pathologic gradients from extremely high values.

You may not have mappability or blacklist tracks for your pig genome, and that's probably OK. But you should be able to track down a BED file of the gaps in the genome assembly where there is no sequence data. I think you can obtain this through UCSC, or you may be able to make it yourself based on looking at N's in your FASTA.

I don't know why you provided 10 targets, but only received accuracy statistics for 3. Could you offer more details about how you ran everything? Can you verify that your TFRecords contain 10 targets?

Predicting 10 million SNPs will take a long time. You can estimate how long by measuring the time required for a sample of your full set, e.g. 10 thousand. I use the script basenji_sad_multi.py to parallelize the computation on our SLURM GPU cluster. You might be able to adjust this script to work for you. Alternatively, you could manually split your VCF file, run the SAD compute in parallel, and combine the results.

**Comment by yinhongwei4079** [comment]
@davek44 Thank you for your kindly reply. we want to put two ATAC-seq samples and eight histone makers(four catogory) samples to train our models and predict the effects of snp across the whole genome. i was not sure for the same markers which have two or more samples we give the same index or different index for the file? we have predict the 1000 snp effects based on our models, it give ten values of SAD, which one we can choose for the effects of snps ? and we want to know the negative value means the snps could be the deleterious  and the positive means begnin? could you give us some advise?

**Comment by davek44** [comment]
I recommend letting each replicate experiment be its own task with its own index.

The ten values for the SNPs correspond to your ten training datasets. The question of which one to study is up to you. If you have no preference, you could look at their average or the first principal component.

The sign of the SNP predictions correspond to the alternative allele minus the reference allele predictions, as you can see here: https://github.com/calico/basenji/blob/master/bin/basenji_sad.py#L359. Thus, negative values indicate that the reference allele is predicted to have greater signal. "Benign" variants that don't have any predicted effect will have values close to zero. SNPs with large positive or negative values can be interesting to study.
